src/main.rs

```diff
@@ -0,0 +1,291 @@
+#![no_std]
+#![no_main]
+#![feature(type_alias_impl_trait)]
+#[allow(arithmetic_overflow)]
+
+// use defmt::*;
+use embassy_executor::Spawner;
+use embassy_stm32::adc::Adc;
+use embassy_stm32::gpio::{Level, Input, Output, Pull, Speed};
+use embassy_stm32::usart::{Config, UartTx};
+use embassy_time::{Timer,Instant,Delay};
+use embassy_stm32::gpio::Level::{High,Low};
+
+use heapless::String;
+use {defmt_rtt as _, panic_probe as _};
+use ringbuffer::{ConstGenericRingBuffer, RingBuffer};
+
+use embassy_sync::signal::Signal;
+use embassy_sync::blocking_mutex::raw::CriticalSectionRawMutex;
+use static_cell::StaticCell;
+
+mod c5412;
+
+pub static C5412PINS_INST: StaticCell<c5412::C5412Pins> = StaticCell::new();
+
+const CRAZY_HI: i32 =3000;
```

Comment:
These all belong with HR, and not the rest of this code.

Maybe call it hr_algorithm?